mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/js/main.js

```diff
@@ -20,7 +28,45 @@ AUI.add(
 					initializer: function(config) {
 						var instance = this;
 
-						var todoList = instance.one('.task-container ul');
+						new A.CharCounter(
+							{
+								counter: '.counter',
+								input: '.input-item',
+								maxLength: 35
+							});
+
+						var todoList = A.one('.task-container ul');
+
+						var finishedTask = A.one('.finished-task');
+
+						var remainingDiv = A.one('.remaining-tasks-count');
+
+						var profileTasksCount = A.one('.profile-tasks-count');
+
+						var taskInput = instance.byId('task');
+
+						var taskList = A.one('.task-list');
+
+						var taskFinished = config.taskFinished;
+
+						instance._taskFinished = taskFinished;
+
+						window.remainingCount = function() {
```

Comment:
Never define anything in global scope, unless some third-party library requires it. Instead, what you need to do is define method as a property of portlet instance js object, just like ```instance._appendTodoList()```. In addition, method name should contain a verb.